src/main/java/org/javarosa/form/api/FormEntryModel.java

```diff
@@ -383,19 +375,20 @@ public boolean isIndexRelevant(FormIndex index) {
         boolean isAskNewRepeat = (getEvent(index) == FormEntryController.EVENT_PROMPT_NEW_REPEAT);
         boolean isRepeatJuncture = (getEvent(index) == FormEntryController.EVENT_REPEAT_JUNCTURE);
 
-        boolean relevant;
         if (isAskNewRepeat) {
-            relevant = form.isRepeatRelevant(ref) && form.canCreateRepeat(ref, index);
+            if (!form.canCreateRepeat(ref, index)) {
+                return false;
+            }
+
+            return form.isRepeatRelevant(ref);
```

Comment:
We most likely need to make a change here to get the test pass. It would be great if we could grab the `GroupDef` for the index/reference and then call something like:

```java
groupDef.isNewRepeatRelevant()
```

This probably means adding the relevance expression to the `GroupDef` when we parse the form.